atlasdb-timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java

```diff
@@ -0,0 +1,170 @@
+        FileUtils.deleteDirectory(new File("var/test/data"));
+    }
+
+    @Test
+    public void lockServiceShouldAllowUsToTakeOutLocks() throws InterruptedException {
+        RemoteLockService lockService = getLockService(CLIENT_1);
+
+        LockRefreshToken token = lockService.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+
+        assertThat(token).isNotNull();
+
+        lockService.unlock(token);
+    }
+
+    @Test
+    public void lockServiceShouldAllowUsToTakeOutSameLockInDifferentNamespaces() throws InterruptedException {
+        RemoteLockService lockService1 = getLockService(CLIENT_1);
+        RemoteLockService lockService2 = getLockService(CLIENT_2);
+
+        LockRefreshToken token1 = lockService1.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+        LockRefreshToken token2 = lockService2.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+
+        assertThat(token1).isNotNull();
+        assertThat(token2).isNotNull();
+
+        lockService1.unlock(token1);
+        lockService2.unlock(token2);
+    }
+
+    @Test
+    public void lockServiceShouldNotAllowUsToRefreshLocksFromDifferentNamespaces() throws InterruptedException {
+        RemoteLockService lockService1 = getLockService(CLIENT_1);
+        RemoteLockService lockService2 = getLockService(CLIENT_2);
+
+        LockRefreshToken token1 = lockService1.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+
+        assertThat(token1).isNotNull();
+        assertThat(lockService1.refreshLockRefreshTokens(ImmutableList.of(token1))).isNotEmpty();
+        assertThat(lockService2.refreshLockRefreshTokens(ImmutableList.of(token1))).isEmpty();
+
+        lockService1.unlock(token1);
+    }
+
+    @Test
+    public void timestampServiceShouldGiveUsIncrementalTimestamps() {
+        TimestampService timestampService = getTimestampService(CLIENT_1);
+
+        long timestamp1 = timestampService.getFreshTimestamp();
+        long timestamp2 = timestampService.getFreshTimestamp();
+
+        assertThat(timestamp1).isLessThan(timestamp2);
+    }
+
+    @Test
+    public void timestampServiceShouldRespectDistinctClientsWhenIssuingTimestamps() {
+        TimestampService timestampService1 = getTimestampService(CLIENT_1);
+        TimestampService timestampService2 = getTimestampService(CLIENT_2);
+
+        long firstServiceFirstTimestamp = timestampService1.getFreshTimestamp();
+        long secondServiceFirstTimestamp = timestampService2.getFreshTimestamp();
+
+        long firstServiceSecondTimestamp = timestampService1.getFreshTimestamp();
+        long secondServiceSecondTimestamp = timestampService2.getFreshTimestamp();
+
+        assertThat(firstServiceFirstTimestamp + 1).isEqualTo(firstServiceSecondTimestamp);
```

Comment:
Unfortunately we don't have a way of reading the timestamp bound easily. We still do cache timestamps but as @harrybiddle has pointed out in the 1 node case (or, actually, as long as you have a consistent leader throughout) this behaviour is safe.

However, I would _not_ include this as a test in say, an abstract integration test for TimeLockServer implementations, since it's kind of specific to this implementation (if you think about it, maintaining a single timestamp service for everyone would satisfy the timestamp service contract).

Changed to use `Assert.assertEquals` (that said, the original AssertJ one I wrote is the wrong way round in terms of expected/actual).